docs/admission_controller.md

```diff
@@ -35,4 +35,65 @@ spec:
 
 The mutated object will contain the following `tls.credentialName=default-httpbin-gateway-https`.
 
-Since the `tls.credentialName` is used to name the `Certificate` and `Secret` resources it is subject to the [253 max character limit](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names).  The `<namespace>-<gateway-name>` will be truncated accordingly to preserve the `portName`
\ No newline at end of file
+Since the `tls.credentialName` is used to name the `Certificate` and `Secret` resources it is subject to the [253 max character limit](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names).  The `<namespace>-<gateway-name>` will be truncated accordingly to preserve the `portName`
+
+The [Controller](./controllers/gateway.md) is responsible for the reconciliation of the referenced `Certificate` and `Secret` resources.
+
+## External DNS Annotation Mutation Logic
+
+- If the controller has external-dns management enabled
+- If the namespace the gateway is created in is subject to mutation
+  - Delete the `external-dns.alpha.kubernetes.io/hostname` annotation if present.
```

Comment:
I think we need to provide more explanation of why the `hostname` annotation is being deleted.